src/Core/InterpolateDescription.cpp

```diff
@@ -30,7 +35,11 @@ namespace DB
             if (const auto & p = aliases.find(name); p != aliases.end())
                 name = p->second->getColumnName();
 
-            result_columns_set.insert(name);
+            auto [it, was_inserted] = result_columns_set.insert(name);
+            if (!was_inserted)
+            {
+                throw Exception(ErrorCodes::BAD_ARGUMENTS, "Duplicate column name '{}' in an interpolate expression", name);
+            }
```

Comment:
this will not work because required column set contains not only column which is interpolated, but also columns from interpolation expression